packages/form/src/reatomField.ts

```diff
@@ -154,176 +197,229 @@ export const fieldInitFocus: FieldFocus = {
-            valid: true,
+            meta: undefined,
+            triggered: true,
             validating: false,
           })
-        },
-      ).catch(noop)
+        }))
 
-      return validationAtom.merge(ctx, {
-        error: keepErrorDuringValidating ? validation.error : undefined,
-        valid: true,
+      return validation.merge({
+        error: keepErrorDuringValidating.value() ? validationValue.error : undefined,
+        meta: undefined,
+        triggered: true,
         validating: true,
       })
     }
 
-    return validationAtom.merge(ctx, {
+    return validation.merge({
       validating: false,
       error: message,
-      valid: true,
+      meta: undefined,
+      triggered: true,
     })
-  }, `${name}.validate`)
+  }, `${name}.validation.trigger`).extend(withAbort())
 
-  const focus: This['focus'] = action((ctx) => {
-    focusAtom.merge(ctx, { active: true })
-  }, `${name}.focus`)
+  const change: This['change'] = action((newValue) => {
+    const prevValue = value()
+    if (!filter(newValue, prevValue)) 
+      return prevValue
 
-  const blur: This['blur'] = action((ctx) => {
-    focusAtom.merge(ctx, { active: false, touched: true })
-  }, `${name}.blur`)
+    field(toState(newValue))
+    focus.merge({ touched: true })
 
-  const change: This['change'] = action((ctx, newValue) => {
-    const prevValue = ctx.get(valueAtom)
-
-    if (!filter(ctx, newValue, prevValue)) return prevValue
-
-    fieldAtom(ctx, toState(ctx, newValue))
-    focusAtom.merge(ctx, { touched: true })
-
-    return ctx.get(valueAtom)
+    return value()
   }, `${name}.change`)
 
-  const reset: This['reset'] = action((ctx) => {
-    fieldAtom(ctx, initState)
-    focusAtom(ctx, fieldInitFocus)
-    validationAtom(ctx, fieldInitValidation)
-    ctx.get(validateControllerAtom).abort(toAbortError('reset'))
-  }, `${name}.reset`)
+  const reset: This['reset'] = action(() => {
+    field(initState())
+    focus(fieldInitFocus)
 
-  if (validateOnChange) {
-    fieldAtom.onChange((ctx) => validate(ctx))
-  }
-
-  if (validateOnBlur) {
-    blur.onCall((ctx) => validate(ctx))
-  }
+    validation(fieldInitValidation)
+    validation.trigger.abort('reset');
+  }, `${name}.reset`)
 
-  return Object.assign(fieldAtom, {
-    blur,
+  return Object.assign(field, {
     change,
     focus,
-    reset,
-    validate,
-
-    focusAtom,
     initState,
-    validationAtom,
-    valueAtom,
+    reset,
+    validation,
+    value,
+    keepErrorDuringValidating,
+    keepErrorOnChange,
+    validateOnChange,
+    validateOnBlur,
+    shouldValidate,
+
+    __reatomField: true as const
   })
 }
+
+export const withField = <T extends Atom, Value = AtomState<T>>(
+  options: Omit<FieldOptions<AtomState<T>, Value>, 'name'> = {}
+): ((anAtom: T) => T & FieldAtom<AtomState<T>, Value>) => {
+  return (anAtom: T) => Object.assign(
+    anAtom,
+    reatomField(null as AtomState<T>, { name: anAtom.name, ...options }, anAtom)
```

Comment:
It isn't worked like this anymore, `reatomField` should probably accept optional "origin" atom for all binds